### PR TITLE
Cleaner test output, enhanced filters.

### PR DESCRIPTION
**Pull request by adityamukho**
### Scope & Purpose

This PR solves 2 issues:
1. The JS test script output is too verbose, containing server startup logs and lifecycle management logs emitted by the process manager that are usually of no use to a tester running normal unit/integration tests on their local machine. They just want to see the test output (pass/fail + stacktrace). A startup command-line option `noStartStopLogs` is added to suppress these extraneous messages. This option is overridden when `extremeVerbosity` is set to `true`.
2. Test case filters were rudimentary, requiring the user to furnish the complete name of a single test case, and even then the script would complain about all the test suites which did not contain that test case. The filters have now been enhanced in the following ways:
    1. All filters now support extended pattern matching, as explained in https://realguess.net/2014/07/05/extended-pattern-matching/.
    2. A filter that contains a '::' separator uses the prefix to filter the test suites and the suffix to filter the test cases within the matching suites. A filter without a '::' separator uses the entire filter expression to match on the test case, as before.
    3. Any suite that does not match the filter condition is silently ignored.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [x] Design document: https://docs.google.com/document/d/13qQpvRondi3d51jZ4dxGoj_lAcfSUAgQXG1pNvBPR2Y/edit#heading=h.uv0ldvt17qkf

### CODE REVIEW

**Comment by dothebart** [comment]
in testing.js:
```
function unitTest (cases, options) {
  if (typeof options !== 'object') {
    options = {};
  }
  loadTestSuites(options);
  // testsuites may register more defaults...
  _.defaults(options, optionsDefaults);
```
appending:
```
options.noStartStopLogs = !options.extremeVerbosity && options.noStartStopLogs;
```
seems nicer to me than having these double-if'ed expressions all over the place...

File: js/client/modules/@arangodb/testutils/testing.js [hunk 1]
```diff
@@ -229,6 +230,7 @@ const optionsDefaults = {
   'valgrindArgs': {},
   'valgrindHosts': false,
   'verbose': false,
+  'noStartStopLogs': false,
```
Comment (dothebart) [inline js/client/modules/@arangodb/testutils/testing.js:233]: Please make this dependend on whether its an interactive shell or not like:
```
  'setInterruptable': ! internal.isATTy(),
```

**Review by dothebart — CHANGES_REQUESTED** [review]
please apply the two suggested simplifications.

File: js/client/modules/@arangodb/testutils/testing.js (hunk 1 above)
Comment (adityamukho) [reply js/client/modules/@arangodb/testutils/testing.js]: Done.

**Comment by adityamukho** [comment]
> in testing.js:
> 
> ```
> function unitTest (cases, options) {
>   if (typeof options !== 'object') {
>     options = {};
>   }
>   loadTestSuites(options);
>   // testsuites may register more defaults...
>   _.defaults(options, optionsDefaults);
> ```
> 
> appending:
> 
> ```
> options.noStartStopLogs = !options.extremeVerbosity && options.noStartStopLogs;
> ```
> 
> seems nicer to me than having these double-if'ed expressions all over the place...

Done.

**Comment by hkernbach** [comment]
> > in testing.js:
> > ```
> > function unitTest (cases, options) {
> >   if (typeof options !== 'object') {
> >     options = {};
> >   }
> >   loadTestSuites(options);
> >   // testsuites may register more defaults...
> >   _.defaults(options, optionsDefaults);
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > appending:
> > ```
> > options.noStartStopLogs = !options.extremeVerbosity && options.noStartStopLogs;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > seems nicer to me than having these double-if'ed expressions all over the place...
> 
> Done.

Just as an idea: Can this also be set with a configuration file or via environment variables?

**Review by dothebart — APPROVED** [review]
please fix this one condition, else LGTM.

File: js/client/modules/@arangodb/testutils/process-utils.js [hunk 2]
```diff
@@ -2178,7 +2202,9 @@ function launchFinalize(options, instanceInfo, startTime) {
                      ' - ' + arangod.args['database.directory']);
   });
 
-  print(Date() + ' sniffing template:\n  tcpdump -ni lo -s0 -w /tmp/out.pcap ' + ports.join(' or ') + '\n');
+  if (!options.noStartStopLogs) {
```
[the remark this answers is not in the document]
Comment (adityamukho) [reply js/client/modules/@arangodb/testutils/process-utils.js]: Removed sniffing template since it is no longer used.

**Review by dothebart — APPROVED** [review]

**Comment by mchacki** [comment]
The reference test has been aborted, and aborted sub-jobs have been restarted.
I ahne validated that Main Job + Restarted sub-Jobs together all have been green, even though the linked PR is declared red here.

**Pull request merged**